packages/plugin-mapbox/src/main/java/com/openmobilehub/android/maps/plugin/mapbox/presentation/maps/OmhMarkerImpl.kt

```diff
@@ -400,8 +407,8 @@ internal class OmhMarkerImpl(
     ): String {
         val markerImageID = getMarkerIconID(icon != null)
 
-        // ensure the other icon is removed for memory optimization
-        style.removeStyleImage(getMarkerIconID(!isCustomIconSet))
+        // ensure the old icon is removed for memory optimization
+        if (lastMarkerIconID != null) style.removeStyleImage(lastMarkerIconID!!)
```

Comment:
```suggestion
        lastMarkerIconID?.let { mapViewDelegate.removeImage(it) }
```

Reply:
Refactored in 3203a66940400da0906cdf14482f0856d617ee7a